sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -288,3 +296,221 @@ def _predict(
         )
 
         return final_predictions
+
+
+class _NeuralForecastAutoAdapter(_NeuralForecastAdapter):
```

Comment:
> One more thing thats blocking me from testing to implementation is due to multilevel inheritance, the `_check_soft_dependancies` util is importing the module/file `neuralforcast.py` instead of package neuralforecast which leads to error `module 'neuralforecast' has no attribute '__version__'`

I am not facing this. I think may be @fkiraly can help with this, but it's very surprising. Multi level inheritance should not cause this issue.

Reply:
> the `_NeuralForecastAdapter` accepts argument `**exogeneous` while instantiating the model. Which is not accepted by auto model in neuralforecast. Is inheriting the base adapter a good idea ?? [here](https://github.com/sktime/sktime/blob/main/sktime/forecasting/base/adapters/_neuralforecast.py#L141)

That's a good point, but I think this is one of the most important features. If `AutoLSTM` is not the auto-hyperparameter-tuning version of `LSTM`, it's very counter-intutive. I think `neuralforecast` does support these (https://github.com/Nixtla/neuralforecast/blob/a6fd29c46e747fa93ed2a661fe475694d236a2a7/neuralforecast/common/_base_auto.py#L412-L415), but passing these to the model does seem difficult/impossible.

@jmoralez @elephaint can you please guide us in the recommended way to pass specific arguments to the underlying models that does not need tuning, e.g. `futr_exog_list` or `num_workers_loader` or configurations for Pythorch Lightning Trainer etc. that needs to be passed to `__init__`, `fit` or `predict` methods of `NeuralForecast` class?

Reply:
You can provide them through the config as fixed arguments, e.g. you could have something like this:
```python
config = {
    "learning_rate": tune.loguniform(1e-5, 1e-1), 
    "futr_exog_list": ["ex1", "ex2"],
    "devices": [0],  # pytorch lightning trainer argument
}
```

Reply:
If I am not wrong, this will then override default_config as of now?

I think users of sktime are unlikely to pass full config to specify exogenous features (it will not be comparable to other models), so may be we have to wait till get_default_config method is released. Or if there's any alternative better way, can you please recommend?

Reply:
Yes, this would become the full config. If you want to keep the default config and just add that then the get_default_config method is probably the best way, we'll try to make the release in the next couple of days.

Reply:
v1.7.0 is now in PyPI, which has the get_default_config method. Please let us know if you encounter any problems with it.